analyticsShared/src/iosMain/kotlin/mealz/ai/SharedAnalytics.ios.kt

```diff
@@ -0,0 +1,108 @@
+    }
+    private val coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.IO + coroutineHandler)
+
+    private val httpClient = HttpClient {
+        install(DefaultRequest)
+    }
+
+    private fun plausiblePropsToDict(props: PlausibleProps): NSDictionary {
+        return mapOf(
+            "recipe_id" to props.recipe_id,
+            "category_id" to props.category_id,
+            "entry_name" to props.entry_name,
+            "item_id" to props.item_id,
+            "ext_item_id" to props.ext_item_id,
+            "item_ean" to props.item_ean,
+            "old_item_id" to props.old_item_id,
+            "old_item_ext_id" to props.old_item_ext_id,
+            "old_item_ean" to props.old_item_ean,
+            "new_item_id" to props.new_item_id,
+            "new_item_ext_id" to props.new_item_ext_id,
+            "new_item_ean" to props.new_item_ean,
+            "product_quantity" to props.product_quantity,
+            "basket_id" to props.basket_id,
+            "miam_amount" to props.miam_amount,
+            "total_amount" to props.total_amount,
+            "miam_products" to props.miam_products,
+            "total_products" to props.total_products,
+            "client_order_id" to props.client_order_id,
+            "pos_id" to props.pos_id,
+            "pos_name" to props.pos_name,
+            "search_term" to props.search_term,
+            "budget" to props.budget,
+            "budget_user" to props.budget_user,
+            "budget_planner" to props.budget_planner,
+            "recipe_count" to props.recipe_count,
+            "guests" to props.guests,
+            "uses_count" to props.uses_count,
+            "time_passed" to props.time_passed,
+            "version" to props.version,
+            "platform" to props.platform,
+            "affiliate" to props.affiliate,
+            "abTestKey" to props.abTestKey
+        ) as NSDictionary
+    }
+
+    @OptIn(ExperimentalForeignApi::class, BetaInteropApi::class)
+    private fun plausibleEventToJson(event: PlausibleEvent): NSString? {
+        val propsDict = plausiblePropsToDict(event.props)
+
+        val eventDict = mapOf(
+            "name" to event.name,
+            "url" to event.url,
+            "domain" to event.domain,
+            "props" to propsDict
+        ) as NSDictionary
+
+        val jsonData: NSData = NSJSONSerialization.dataWithJSONObject(eventDict, 0u, null)!!
+
+        return NSString.create(data = jsonData, encoding = NSUTF8StringEncoding)
+    }
+
+    private suspend fun HttpClient.postEvent(event: PlausibleEvent) {
+        println("will send event $event to $PLAUSIBLE_URL") // TODO: log and log levels task
+        post(PLAUSIBLE_URL) {
+            contentType(ContentType.Application.Json)
+            setBody(plausibleEventToJson(event))
+        }
+    }
+
+    override fun sendRequest(event: PlausibleEvent) {
+        coroutineScope.launch {
+            httpClient.postEvent(event)
```

Comment:
on a besoin de deux fonctions ?

Reply:
bonne question, ktor est nouveau pour moi, j'ai copié collé le code existant

Reply:
je pense que ça fait le travail :
```suggestion
            httpClient. post(PLAUSIBLE_URL) {
            contentType(ContentType.Application.Json)
            setBody(plausibleEventToJson(event))
        }
```